bump_pydantic/codemods/validator.py

```diff
@@ -155,7 +166,13 @@ def _decorator_with_leading_comment(self, node: cst.Decorator, comment: str) ->
     def _replace_validators(self, node: cst.Decorator, old_name: str, new_name: str) -> cst.Decorator:
         RemoveImportsVisitor.remove_unused_import(self.context, "pydantic", old_name)
         AddImportsVisitor.add_needed_import(self.context, "pydantic", new_name)
-        decorator = node.decorator.with_changes(func=cst.Name(new_name), args=self._args)
+        if m.matches(node.decorator, m.Call()):
+            decorator = node.decorator.with_changes(func=cst.Name(new_name), args=self._args)
+        else:
+            # Assuming node.decorator matches Name()
+            # (once ROOT_VALIDATOR_DECORATOR supports Attribute() this needs a separate case).
+            # Changing `@root_validator` (Name, not Call) into `@model_validator(...)` (Call)
+            decorator = cst.Call(func=cst.Name(new_name), args=self._args)
```

Comment:
But it shouldn't be an else, should it? Does the test you mention fails differently now?

Reply:
If I make it an `elif` (like before) without the `else` branch, mypy complains about `decorator` potentially being unbound.

> Does the test you mention fails differently now?

Well, the test fails the same everytime ATM, because we never reach that case due to what `ROOT_VALIDATOR_DECORATOR` is matching (it's not matching decorators as attributes now, so those cases won't be processed even).